ethereum/geth_client_integration_test.go

```diff
@@ -93,8 +93,22 @@ func TestGetConnection_returnsSameConnection(t *testing.T) {
 	}
 }
 
-func TestNewGethClient(t *testing.T) {
+func TestGethClient_GetTxOpts(t *testing.T) {
 	gc, err := ethereum.NewGethClient(cfg)
-	assert.Nil(t, err)
+	assert.NoError(t, err)
 	assert.NotNil(t, gc)
+
+	opts, err := gc.GetTxOpts(context.Background(), "main")
+	assert.NoError(t, err)
+	assert.True(t, opts.GasPrice.Cmp(big.NewInt(20000000000)) == 0)
+
+	cfg.Set("ethereum.maxGasPrice", 10000000000)
+	gc, err = ethereum.NewGethClient(cfg)
+	assert.NoError(t, err)
+	assert.NotNil(t, gc)
+	opts, err = gc.GetTxOpts(context.Background(), "main")
+	if assert.Error(t, err) {
```

Comment:
we want to make sure that an error is triggered in this scenario, can we force it, instead of having a if clause?

Reply:
it is forced in this case. its the recommended way to assert an error :)